### PR TITLE
Refactored Dropbox

### PR DESCRIPTION
**Pull request by johnathan-rodriguez**

### CODE REVIEW

File: shipyard_blueprints/dropbox/shipyard_dropbox/dropbox.py [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
-        client = Dropbox(self.access_key)
-        client.users_get_current_account()
-        return client
+        try:
+            Dropbox(self.access_key).users_get_current_account()
+            logger.authtest("Successfully verified Dropbox access key.")
+            return 0
+        except Exception as e:
+            logger.authtest(
+                f"Failed to authenticate using key. Message from Dropbox Server{e}"
```
Comment (wrp801) [inline shipyard_blueprints/dropbox/shipyard_dropbox/dropbox.py:18]: Change the message to this for slightly easier readability:

```f"Failed to authenticate using key. Message from Dropbox Server: {e}"```

File: shipyard_blueprints/dropbox/shipyard_dropbox/cli/download.py [hunk 2]
```diff
@@ -179,77 +86,77 @@ def download_dropbox_file(file_name, client, destination_file_name=None):
...
+                    source_full_path=file_name,
+                    file_number=index,
+                )
+
+                logger.info(f"Downloading file {index} of {number_of_matching_files}")
+                download_dropbox_file(
+                    file_name=(
+                        file_name if file_name.startswith("/") else f"/{file_name}"
+                    ),
+                    client=client,
```
Comment (wrp801) [inline shipyard_blueprints/dropbox/shipyard_dropbox/cli/download.py:152]: For consistency, upon completion have a log statement that says "Successfully downloaded file to <target_path>". For the `regex_match` you don't need to do that for each one, but you can say something like "Successfully downloaded all files"

File: shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py [hunk 3]
```diff
@@ -171,85 +78,87 @@ def download_dropbox_file(file_name, client, destination_file_name=None):
...
 
-        if not source_full_path.startswith("/"):
-            source_full_path = f"/{source_full_path}"
-        download_dropbox_file(
-            file_name=source_full_path,
-            client=client,
-            destination_file_name=destination_name,
-        )
+            logger.info(
+                f"{len(matching_file_names)} files found. Preparing to download..."
```
Comment (wrp801) [inline shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py:140]: Should be "Preparing to upload" instead

File: shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py [hunk 4]
```diff
@@ -171,85 +78,87 @@ def download_dropbox_file(file_name, client, destination_file_name=None):
...
+
+                logger.info(f"Uploading file {index} of {number_of_matching_files}")
+                download_dropbox_file(
+                    file_name=(
+                        file_name if file_name.startswith("/") else f"/{file_name}"
+                    ),
+                    client=client,
+                    destination_file_name=destination_name,
+                )
+    except Exception as e:
```
Comment (wrp801) [inline shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py:159]: Same comment as before, where upon completion have one more additional log statement that lets the user know that their file has been successfully uploaded to Dropbox

**Review by wrp801 — CHANGES_REQUESTED** [review]
Small changes around the log statements

**Comment by johnathan-rodriguez** [comment]
This is one of the blueprints that refer to the main branch but looks like it is currently broken since the download function was carried over twice. Blueprint is currently not being used. Should I point it back to the original https://github.com/shipyardapp/dropbox-blueprints repo

**Comment by wrp801** [comment]
This PR has a lot of changes to other vendors that are most likely the result of a stale branch. Try pulling in the latest commits from main and merging them to this

**Pull request closed without merging**